[PyTorch] Add input numel assert for minimal arrayref interface

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #113578
* __->__ #113577
* #112800

We currently have no shape checking on CPU IIUC. Now we at least do numel checking for the minimal arrayref interface.

Differential Revision: [D51165703](https://our.internmc.facebook.com/intern/diff/D51165703/)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

swolchok committed:
[PyTorch] Add input numel assert for minimal arrayref interface

We currently have no shape checking on CPU IIUC. Now we at least do numel checking for the minimal arrayref interface.

Differential Revision: [D51165703](https://our.internmc.facebook.com/intern/diff/D51165703/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add input numel assert for minimal arrayref interface"


We currently have no shape checking on CPU IIUC. Now we at least do numel checking for the minimal arrayref interface.

Differential Revision: [D51165703](https://our.internmc.facebook.com/intern/diff/D51165703/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add input numel assert for minimal arrayref interface"


We currently have no shape checking on CPU IIUC. Now we at least do numel checking for the minimal arrayref interface.

Differential Revision: [D51165703](https://our.internmc.facebook.com/intern/diff/D51165703/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add input numel assert for minimal arrayref interface"


We currently have no shape checking on CPU IIUC. Now we at least do numel checking for the minimal arrayref interface.

Differential Revision: [D51165703](https://our.internmc.facebook.com/intern/diff/D51165703/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add input numel assert for minimal arrayref interface"


We currently have no shape checking on CPU IIUC. Now we at least do numel checking for the minimal arrayref interface.

Differential Revision: [D51165703](https://our.internmc.facebook.com/intern/diff/D51165703/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add input numel assert for minimal arrayref interface"


We currently have no shape checking on CPU IIUC. Now we at least do numel checking for the minimal arrayref interface.

Differential Revision: [D51165703](https://our.internmc.facebook.com/intern/diff/D51165703/)

[ghstack-poisoned]

chenyang78 commented on torch/_inductor/codegen/wrapper.py:
```diff
@@ -1624,12 +1624,27 @@ def write_wrapper_decl(self):
+                continue
+
+            # comparing strides for 0 size tensor is tricky. Ignore them for now.
+            if sympy_product(buf.get_size()) == 0:
```
Curious - for 0-size-tensor cases, why can't we have something like ```assert_num(buf_name, 0)```?

chenyang78 reviewed (approved):
Left a minor comment. Please fix the linter failure. LGTM otherwise. Thanks!

swolchok committed:
Update on "[PyTorch] Add input numel assert for minimal arrayref interface"


We currently have no shape checking on CPU IIUC. Now we at least do numel checking for the minimal arrayref interface.

Differential Revision: [D51165703](https://our.internmc.facebook.com/intern/diff/D51165703/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add input numel assert for minimal arrayref interface"


We currently have no shape checking on CPU IIUC. Now we at least do numel checking for the minimal arrayref interface.

Differential Revision: [D51165703](https://our.internmc.facebook.com/intern/diff/D51165703/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add input numel assert for minimal arrayref interface"


We currently have no shape checking on CPU IIUC. Now we at least do numel checking for the minimal arrayref interface.

Differential Revision: [D51165703](https://our.internmc.facebook.com/intern/diff/D51165703/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add input numel assert for minimal arrayref interface"


We currently have no shape checking on CPU IIUC. Now we at least do numel checking for the minimal arrayref interface.

Differential Revision: [D51165703](https://our.internmc.facebook.com/intern/diff/D51165703/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add input numel assert for minimal arrayref interface"


We currently have no shape checking on CPU IIUC. Now we at least do numel checking for the minimal arrayref interface.

Differential Revision: [D51165703](https://our.internmc.facebook.com/intern/diff/D51165703/)

[ghstack-poisoned]
